src/xdg.rs

```diff
@@ -74,25 +78,42 @@ impl DesktopEntries {
                     Err(_) => continue,
                 };
 
+                // Ignore all groups other than the `Desktop Entry` one.
+                //
+                // Since `Desktop Entry` must be the first group, we just stop at the next group
+                // header.
+                let lines = desktop_file.lines().take_while(|line| {
+                    line.trim_end() == "[Desktop Entry]" || !line.starts_with('[')
+                });
```

Comment:
This is just a minor simplification of what you already implemented. It should be functionally identical since `[Desktop Entry]` is required to be the first group in the file according to the spec.